src/ripple/protocol/impl/Feature.cpp

```diff
@@ -18,57 +18,73 @@
 //==============================================================================
 
 #include <BeastConfig.h>
+#include <ripple/protocol/Feature.h>
 #include <ripple/basics/contract.h>
 #include <ripple/protocol/digest.h>
-#include <ripple/protocol/Feature.h>
 
 #include <boost/container/flat_map.hpp>
 
 #include <cstring>
 
 namespace ripple {
 
-static
-uint256
-feature (char const* s, std::size_t n)
-{
-    sha512_half_hasher h;
-    h(s, n);
-    return static_cast<uint256>(h);
-}
-
-static
-uint256
-feature (std::string const& name)
-{
-    return feature(name.c_str(), name.size());
-}
-
-static std::vector<uint256> const features = [] {
-    std::vector<uint256> result;
-    result.reserve(detail::numFeatures);
-    for (std::string const& s : detail::featureNames)
-        result.push_back(feature(s));
-    return result;
-}();
+static constexpr char const* const featureNames[] =
+    {"MultiSign",
+     "Tickets",
+     "TrustSetAuth",
+     "FeeEscalation",
+     "OwnerPaysFee",
+     "CompareFlowV1V2",
+     "SHAMapV2",
+     "PayChan",
+     "Flow",
+     "CompareTakerFlowCross",
+     "FlowCross",
+     "CryptoConditions",
+     "TickSize",
+     "fix1368",
+     "Escrow",
+     "CryptoConditionsSuite",
+     "fix1373",
+     "EnforceInvariants"};
+
+static_assert (detail::numFeatures() ==
+    sizeof (featureNames) / sizeof (featureNames[0]), "Update numFeatures");
```

Comment:
I want to minimize the number of changes needed when we add a new feature, and I want those changes to be as localized as possible. If we could automatically compute `detail::numFeatures`, instead of the `static_assert` that would be a win.